src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,692 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include "info_aie.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace qr = xrt_core::query;
+using ptree_type = boost::property_tree::ptree;
+
+const uint32_t major = 1;
+const uint32_t minor = 0;
+const uint32_t patch = 0;
+
+enum graph_state {
+  STOP = 0,
+  RESET = 1,
+  RUNNING = 2,
+  SUSPEND = 3,
+  END = 4
+};
+
+namespace {
+
+inline std::string
+graph_status_to_string(int status) {
+  switch(status)
+  {
+    case STOP:    return std::string("stop");
+    case RESET:   return std::string("reset");
+    case RUNNING: return std::string("running");
+    case SUSPEND: return std::string("suspend");
+    case END:     return std::string("end");
+    default:      return std::string("idle");
+  }
+}
+
+static void
+addnodelist(std::string search_str, std::string node_str,boost::property_tree::ptree& input_pt, boost::property_tree::ptree& output_pt)
```

Comment:
Either take advantage of the copied argument string and move it to its destination or change all to const std::string&

Reply:
Changed all the variable to const. Please check . Thanks